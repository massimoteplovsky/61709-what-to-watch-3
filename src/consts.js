export const DEFAULT_GENRE = `All genres`;
export const FILM_TO_SHOW = 8;
export const RELATED_FILM_COUNT = 4;
export const AUTH = `AUTH`;
export const NO_AUTH = `NO_AUTH`;
export const BASE_URL = `https://htmlacademy-react-2.appspot.com`;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const NOT_FAVORITE_FILM = 0;
export const FAVORITE_FILM = 1;
export const MAX_RATING = 5;
