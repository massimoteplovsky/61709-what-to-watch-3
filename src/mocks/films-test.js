export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: [`Action`],
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 2,
    title: `1+1`,
    genre: [`Drama`, `Comedy`],
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 3,
    title: `Joker`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 6,
    title: `1917`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 7,
    title: `Parasite`,
    genre: [`Drama`, `Comedy`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 8,
    title: `Irishman`,
    genre: [`drama`, `criminal`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: [`Action`],
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: []
};

export const promoFilmInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
