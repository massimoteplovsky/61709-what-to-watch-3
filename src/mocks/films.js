export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
