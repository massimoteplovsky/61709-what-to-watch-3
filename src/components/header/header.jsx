import React from "react";
import {connect} from "react-redux";
import {NO_AUTH, BASE_URL} from "../../consts.js";
import {PropValidator} from "../../prop-validator/prop-validator.js";
import {PropTypes} from "prop-types";
import {Link} from "react-router-dom";
import history from "../../history.js";
import withActiveItem from "../../hocs/with-active-item/with-active-item.js";
import {getAuthorizationStatus} from "../../selectors/user/user.js";

const Header = ({authorizationStatus, userInfo, additionalClass}) => {

  return (
    <header className={`page-header ${additionalClass}`}>
      <div className="logo">
        <Link className="logo__link" to={`/`}>
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      <div className="user-block">
        {
          authorizationStatus === NO_AUTH ?
            <Link className="user-block__link" to={`/login`}>Sign in</Link>
            :
            <div className="user-block__avatar">
              <img
                src={userInfo ? `${BASE_URL}${userInfo.avatarUrl}` : null}
                alt={userInfo ? userInfo.name : null} width="63" height="63"
                onClick={() => history.push(`/myList`)}
              />
            </div>
        }
      </div>
    </header>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  userInfo: state.user.userInfo
});

Header.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  userInfo: PropValidator.USER_INFO,
  additionalClass: PropTypes.string.isRequired
};

export {Header};
export default connect(mapStateToProps)(withActiveItem(Header));
