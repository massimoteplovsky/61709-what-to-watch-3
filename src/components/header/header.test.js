import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header.jsx";
import {userInfo} from "../../mocks/films-test";
import {Router} from "react-router-dom";
import {AUTH} from "../../consts.js";
import history from "../../history";

it(`<Header /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Header
            authorizationStatus={AUTH}
            userInfo={userInfo}
            additionalClass={``}
          />
        </Router>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
