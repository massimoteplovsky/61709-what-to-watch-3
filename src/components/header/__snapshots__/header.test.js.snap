// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component renders correctly 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt=""
        height="63"
        onClick={[Function]}
        src="https://htmlacademy-react-2.appspot.com"
        width="63"
      />
    </div>
  </div>
</header>
`;
